WordPress/gutenberg
# Site Editor: Wrap each router area in 'ErrorBoundary'

## What?
PR moves the Site Editor error boundary up to the React tree to catch errors for every route.

## Why?
The error triggered outside the block editor canvas was causing a "grey screen of death".

## Testing Instructions
1. Trigger an error outside the canvas.
2. Navigate to `/wp-admin/site-editor.php?postType=wp_template`.
3. Confirm that the error boundary is correctly displayed.

### Testing Instructions for Keyboard
Same.

## Screenshots or screencast 
![CleanShot 2024-08-05 at 18 17 05](https://github.com/user-attachments/assets/6a3a3c64-60be-49dd-83c6-1910fbf4c7a9)
